src/seeds.ts

```diff
@@ -0,0 +1,29 @@
+import { AppDataSource } from './data-source.js';
+import { User } from './entity/User.js';
+import { setupDatabase } from './setup-database.js';
+import bcrypt from 'bcrypt';
+import faker from 'faker';
+
+const NUM_USERS = 50;
+
+async function seedDatabase() {
+  try {
+    await setupDatabase();
+    const userRepository = AppDataSource.getRepository(User);
+
+    for (let i = 0; i < NUM_USERS; i++) {
+      userRepository.save({
+        name: faker.name.findName(),
+        email: faker.internet.email(),
+        password: await bcrypt.hash(faker.internet.password(), 10),
+        birthDate: faker.date.past().toISOString().slice(0, 10),
+      });
+    }
```

Comment:
2 comentários sobre esse trecho:

1. Aqui vc está salvando um usuário de cada vez, fazendo assim, 50 queries pro banco. Só o que `.save` aceita também arrays, então vc consegue fazer apenas uma query e salvar os 50 usuários de uma vez, fica bem mais eficiente
2. O `.save` retorna uma Promise, porém como não tem um await na utilização dele, ele vai acabar iniciando a execução das Promises, mas não necessariamente vai finalizar. E portanto pode ser que ocorra algum erro no `.save` que não caia no catch, porque a execução da função `seedDatabase` pode já ter terminado antes de uma ou das várias promises dentro do for resolverem. Faz sentido?